# PTO - Predicting the Oscars


This notebook covers my progress on building an algorithm to accurately predict Oscar results. Not just the winners based on the nominees in each category, but also which movies will be nominees based on all of the movies released that year. The files can be cloned at https://github.com/j0hnk1m/predict-the-oscars if you're interested.

This mini-project only lasted for a month, so it may not be perfect. Hopefully, you guys can use this as a platform to get even better predictions. Let's dig in!

## 1. Data Collection

First things first, general imports first

In [3]:
import pandas as pd
import numpy as np
import os

I came across this dataset from https://bigml.com/user/academy_awards/gallery/dataset/5c6886e1eba31d73070017f5, which contained a list of movies from 2000~2018 and their details, including release year, movie_id (IMDB), certificate, duration, genre, IMDB rating, etc.

Here's what it looks like:

In [8]:
bigml = pd.read_csv('./data/BigML_Dataset_5ccb3e8ddb8b1d0886002d22.csv')
print(bigml.shape)
bigml.head(5)

(1235, 119)


year                                              movie   movie_id  \
0  2001                                     Kate & Leopold  tt0035423   
1  2000                                        Chicken Run  tt0120630   
2  2005                                     Fantastic Four  tt0120667   
3  2002                                              Frida  tt0120679   
4  2001  The Lord of the Rings: The Fellowship of the Ring  tt0120737   

  certificate  duration                       genre  rate  metascore  \
0       PG-13       118      Comedy|Fantasy|Romance   6.4       44.0   
1           G        84  Animation|Adventure|Comedy   7.0       88.0   
2       PG-13       106     Action|Adventure|Family   5.7       40.0   
3           R       123     Biography|Drama|Romance   7.4       61.0   
4       PG-13       178     Adventure|Drama|Fantasy   8.8       92.0   

                                            synopsis    votes  ...  \
0  An English Duke from 1876 is inadvertedly drag...    66660  ...   
1  When a cockerel apparently flies into a chicke...   144475  ...   
2  A group of astronauts gain superpowers after a...   273203  ...   
3  A biography of artist Frida Kahlo, who channel...    63852  ...   
4  A meek Hobbit from the Shire and eight compani...  1286275  ...   

   New_York_Film_Critics_Circle_nominated  \
0                                       0   
1                                       1   
2                                       0   
3                                       0   
4                                       0   

  New_York_Film_Critics_Circle_nominated_categories  \
0                                               NaN   
1                                Best Animated Film   
2                                               NaN   
3                                               NaN   
4                                               NaN   

   Los_Angeles_Film_Critics_Association_won  \
0                                         0   
1                                         1   
2                                         0   
3                                         0   
4                                         1   

   Los_Angeles_Film_Critics_Association_won_categories  \
0                                                NaN     
1                                     Best Animation     
2                                                NaN     
3                                                NaN     
4                                         Best Music     

   Los_Angeles_Film_Critics_Association_nominated  \
0                                               0   
1                                               1   
2                                               0   
3                                               0   
4                                               2   

   Los_Angeles_Film_Critics_Association_nominated_categories  \
0                                                NaN           
1                                     Best Animation           
2                                                NaN           
3                                                NaN           
4                  Best Music|Best Production Design           

   release_date.year release_date.month release_date.day-of-month  \
0             2001.0               12.0                      25.0   
1             2000.0                6.0                      23.0   
2             2005.0                7.0                       8.0   
3             2002.0               11.0                      22.0   
4             2001.0               12.0                      19.0   

  release_date.day-of-week  
0                      2.0  
1                      5.0  
2                      5.0  
3                      5.0  
4                      3.0  

[5 rows x 119 columns]

And each of the 1235 movies in this dataset has a lot of variables (119). If we print all of them out:

In [9]:
print(bigml.columns)

Index(['year', 'movie', 'movie_id', 'certificate', 'duration', 'genre', 'rate',
       'metascore', 'synopsis', 'votes',
       ...
       'New_York_Film_Critics_Circle_nominated',
       'New_York_Film_Critics_Circle_nominated_categories',
       'Los_Angeles_Film_Critics_Association_won',
       'Los_Angeles_Film_Critics_Association_won_categories',
       'Los_Angeles_Film_Critics_Association_nominated',
       'Los_Angeles_Film_Critics_Association_nominated_categories',
       'release_date.year', 'release_date.month', 'release_date.day-of-month',
       'release_date.day-of-week'],
      dtype='object', length=119)
